src/CMakeLists.txt

```diff
@@ -211,9 +211,13 @@ if(AV_EIGEN_MEMORY_ALIGNMENT)
     AddCompilerFlag("-faligned-new")
   elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 6.0)
     AddCompilerFlag("-faligned-new")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.0)
+    AddCompilerFlag("-faligned-new")
```

Comment:
I'm wondering if these flags for eigen memory alignment shouldn't be a property of each target instead rather than added like this.
I'm not sure that if the library is then used as a 3rd party the flag added with AddCompilerFlag will be propagated.
Usually, if we add a flag this way (take -Wall as an example) it only applies for the current build, but if the library, once built, is used to link with another project the flag is not added automatically. It's up to the host project. In the case of warning is not an issue, but for flags that change the behavior of the code, it could.

And sadly we have a lot of template code in AV which could be compiled again by the other project. So potentially the other project is compiling this code with a different set of flags and at the same time using also compiled code of the library with another set of flags. That could turn ugly.
Any thoughts?

Reply:
I agree in principle, but I think in this particular case the work needed would outweigh the benefit due to the following reasons:
 - this flag is enabled by the user of aliceVision via `AV_EIGEN_MEMORY_ALIGNMENT`. Presumably the user knows what it does and will adjust their own project flags accordingly.
 - this flag is only a temporary solution until we can default to C++17.